lib/vanagon/driver.rb

```diff
@@ -31,11 +31,12 @@ def initialize(platform, project, options = { :configdir => nil, :target => nil,
       engine = 'base' if target
       # Hardware has explicit teardown to unlock the node
       engine = 'hardware' if @platform.build_hosts
+      engine = 'docker' if @platform.docker_image
+      engine = 'amazon' if @platform.aws_ami
+
       require "vanagon/engine/#{engine}"
       @engine = Object.const_get("Vanagon::Engine::#{engine.capitalize}").new(@platform, target)
 
-    rescue LoadError => e
```

Comment:
I recall having similar problems when adding the HW engine. I wonder if that commit got lost or if I never fixed it.